src/Transform/ONNX/ReplaceNoValueOperands.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===--------- ReplaceNoValueOperands.cpp - ReplaceNoValueOperands Op------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file replace onnx.NoValue ops with the standard value constant
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include "src/Dialect/ONNX/ONNXDialect.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/Dialect/ONNX/ONNXOps/OpHelper.hpp"
+#include "src/Pass/Passes.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+namespace {
+struct ReplaceNoValuePass : public mlir::PassWrapper<ReplaceNoValuePass,
+                                OperationPass<mlir::ModuleOp>> {
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(ReplaceNoValuePass)
+
+  ReplaceNoValuePass() = default;
+  ReplaceNoValuePass(const ReplaceNoValuePass &pass)
+      : PassWrapper<ReplaceNoValuePass, OperationPass<ModuleOp>>() {}
+
+  StringRef getArgument() const override { return "onnx-replace-novalue"; }
+
+  StringRef getDescription() const override {
+    return "Replaces to NoValue op with an constant op of the standard shape "
```

Comment:
```suggestion
    return "Replaces NoValue op with a constant op of the standard shape "
```

Reply:
Also what do you mean by "standard shape and value"

Reply:
I mean "standard shape and value" as defined in the onnx docs for the operation

Reply:
Right, not so fond of the `standard` here, maybe default is better? But that's just nitpicking, feel free to do as you wish